--- output/gopurs_runtime/runtime.go
+++ output/gopurs_runtime/runtime.go
@@ -3,6 +3,7 @@
 import (
 	"fmt"
 	"math"
+	"runtime/debug"
 	"unsafe"
 )
 
@@ -474,6 +475,10 @@
 
 func Apply(f Value, arg Value) Value {
 	if f.Type != TypeFunc1 && f.Type != TypeFunc2 && f.Type != TypeFunc3 && f.Type != TypeFunc4 && f.Type != TypeFunc5 && f.Type != TypeFunc6 && f.Type != TypeFunc7 && f.Type != TypeFunc8 && f.Type != TypeFunc9 && f.Type != TypeFunc10 {
+		if f.Type == TypeBool {
+			fmt.Printf("[DEBUG] TypeBool passed to Apply! f=%+v\n", f)
+			fmt.Printf("%s\n", debug.Stack())
+		}
 		panic(fmt.Sprintf("Attempted to apply a non-function: f=%#v arg=%#v", f, arg))
 	}
 	switch f.Type {
