--- src/Effect/Aff.go
+++ src/Effect/Aff.go
@@ -43,7 +43,7 @@
 	return func(ctx context.Context) (any, error) {
 		select {
 		case <-ctx.Done():
-			return nil, ctx.Err()
+			return nil, context.Cause(ctx)
 		default:
 			return val, nil
 		}
@@ -66,7 +66,7 @@
 		case <-timer.C:
 			return nil, nil
 		case <-ctx.Done():
-			return nil, ctx.Err()
+			return nil, context.Cause(ctx)
 		}
 	}
 }
@@ -75,7 +75,7 @@
 	return func(ctx context.Context) (any, error) {
 		select {
 		case <-ctx.Done():
-			return nil, ctx.Err()
+			return nil, context.Cause(ctx)
 		default:
 			return eff(nil), nil
 		}
@@ -118,7 +118,7 @@
 		case <-ctx.Done():
 			cancelFnAff := canceler(fmt.Errorf("context canceled"))
 			go runAffSync(cancelFnAff, context.Background())
-			return nil, ctx.Err()
+			return nil, context.Cause(ctx)
 		}
 	}
 }
@@ -259,7 +259,7 @@
 				}
 				return resultVal, nil
 			} else {
-				if ctx.Err() != nil && context.Cause(ctx) == err {
+				if context.Cause(ctx) != nil && context.Cause(ctx) == err {
 					return nil, err
 				}
 				return nil, err
@@ -366,7 +366,7 @@
 		case res := <-resultChan:
 			return res.val, res.err
 		case <-ctx.Done():
-			return nil, ctx.Err()
+			return nil, context.Cause(ctx)
 		}
 	}
 }
@@ -401,7 +401,7 @@
 		
 		if useErr != nil {
 			var cleanupVal gopurs_runtime.Value
-			if ctx.Err() != nil && context.Cause(ctx) == useErr {
+			if context.Cause(ctx) != nil && context.Cause(ctx) == useErr {
 				errBox := gopurs_runtime.Box(useErr)
 				cleanupVal = gopurs_runtime.Apply2(killedVal, errBox, gopurs_runtime.Box(resourceBox))
 			} else {
