--- src/Effect/Aff.go
+++ src/Effect/Aff.go
@@ -179,7 +179,7 @@
 }
 
 
-func _RunFiber(nf *NativeFiber, _ interface{}) any {
+func _RunFiber(nf *NativeFiber) any {
 	return func(_ any) any {
 		select {
 		case <-nf.Start:
@@ -233,8 +233,10 @@
 }
 }
 
-func _IsSuspendedFiber(nf *NativeFiber) bool {
-	return false
+func _IsSuspendedFiber(nf *NativeFiber) any {
+	return func(_ any) any {
+		return false
+	}
 }
 
 func _ThrowError(err error) any {
