--- test/Test/Main.purs
+++ test/Test/Main.purs
@@ -255,8 +255,8 @@
   f3 <- forkAff $ bracketAction "baz" (const (action "c"))
   r3 <- try $ joinFiber f3
 
-  r4 <- readRef ref
-  Console.log ("r4 is: " <> r4)
+  r4 <- liftEffect $ readRef ref
+  liftEffect $ Console.log ("r4 is: " <> r4)
   pure (isLeft r1 && isLeft r2 && isRight r3 && r4 == "foofoo/kill/zbarbar/throw/bbazcbaz/release/c")
 
 test_supervise :: Aff Unit
@@ -552,7 +552,7 @@
           <|> action 16.0 "f"
       )
   delay (Milliseconds 20.0)
-  r4 <- readRef ref
+  r4 <- liftEffect $ readRef ref
   pure (r1 == "a" && r2 == "b" && r3 == "de" && r4 == "abde")
 
 test_kill_parallel_alt :: Aff Unit
