--- output/Effect.Aff/Effect_Aff_ffi.go
+++ output/Effect.Aff/Effect_Aff_ffi.go
@@ -633,6 +633,10 @@
 })
 var _Gopurs__LiftEffect = // TAST: (Func [(ADT ["Effect","Effect"] [(TypeVar a)])] (ADT ["Effect","Aff","Aff"] [(TypeVar a)]))
 gopurs_runtime.Func(func(arg0 gopurs_runtime.Value) gopurs_runtime.Value {
+	if arg0.Type == gopurs_runtime.TypeBool {
+		import_fmt.Printf("[DEBUG] _LiftEffect called with TypeBool: %+v\n", arg0)
+		panic("_LiftEffect with TypeBool")
+	}
 	go_arg0 := func(p0_0 any) any {
 			return gopurs_runtime.Apply(arg0, gopurs_runtime.Box(p0_0))
 		}
